packages/@aws-cdk/aws-glue-alpha/lib/jobs/scala-spark-streaming-job.ts

```diff
@@ -0,0 +1,189 @@
+    this.sparkUILoggingLocation = sparkUIArgs?.location;
+
+    // Enable CloudWatch metrics and continuous logging by default as a best practice
+    const continuousLoggingArgs = props.continuousLogging?.enabled ? this.setupContinuousLogging(this.role, props.continuousLogging) : {};
+    const profilingMetricsArgs = { '--enable-metrics': '' };
+
+    // Gather executable arguments
+    const executableArgs = this.executableArguments(props);
+
+    // Mandatory className argument
+    if (props.className === undefined) {
+      throw new Error('className must be set for Scala ETL Jobs');
+    }
+
+    // Conbine command line arguments into a single line item
+    const defaultArguments = {
+      ...executableArgs,
+      ...continuousLoggingArgs,
+      ...profilingMetricsArgs,
+      ...sparkUIArgs?.args,
+      ...this.checkNoReservedArgs(props.defaultArguments),
+    };
+
+    if ((!props.workerType && props.numberOrWorkers !== undefined) || (props.workerType && props.numberOrWorkers === undefined)) {
+      throw new Error('Both workerType and numberOrWorkers must be set');
+    }
+
+    const jobResource = new CfnJob(this, 'Resource', {
+      name: props.jobName,
+      description: props.description,
+      role: this.role.roleArn,
+      command: {
+        name: JobType.STREAMING,
+        scriptLocation: this.codeS3ObjectUrl(props.script),
+      },
+      glueVersion: props.glueVersion ? props.glueVersion : GlueVersion.V4_0,
+      workerType: props.workerType ? props.workerType : WorkerType.G_2X,
+      numberOfWorkers: props.numberOrWorkers ? props.numberOrWorkers : 10,
+      maxRetries: props.maxRetries,
+      executionProperty: props.maxConcurrentRuns ? { maxConcurrentRuns: props.maxConcurrentRuns } : undefined,
+      timeout: props.timeout?.toMinutes(),
+      connections: props.connections ? { connections: props.connections.map((connection) => connection.connectionName) } : undefined,
+      securityConfiguration: props.securityConfiguration?.securityConfigurationName,
+      tags: props.tags,
+      defaultArguments,
+    });
+
+    const resourceName = this.getResourceNameAttribute(jobResource.ref);
+    this.jobArn = this.buildJobArn(this, resourceName);
+    this.jobName = resourceName;
+  }
+
+  /**
+   * Set the executable arguments with best practices enabled by default
+   *
+   * @param props
+   * @returns An array of arguments for Glue to use on execution
+  */
+  private executableArguments(props: ScalaSparkStreamingJobProps) {
+    const args: { [key: string]: string } = {};
+    args['--job-language'] = JobLanguage.SCALA;
+    args['--class'] = props.className!;
+
+    // TODO: Confirm with Glue service team what the mapping is from extra-x to job language, if any
+    if (props.extraJars && props.extraJars?.length > 0) {
+      // args['--extra-jars'] = props.extraJars.map(code => this.codeS3ObjectUrl(code)).join(',');
```

Comment:
uncomment - if the developer provides extra jars we need to add them to the arguments.

Reply:
Remove TODO comment.